File: abc152/e.cpp
#include <bits/stdc++.h>
using namespace std;
using ll = long long;
constexpr ll MOD = 1e9+7;

ll gcd(ll a, ll b) {
	if(b == 0) return a;
	if(b > a) return gcd(b,a);
	return gcd(b, a%b);
}

ll lcm(ll a, ll b) {
	return a * b / gcd(a,b);
}

int main(){
	int N;
	cin >> N;
	vector<ll> A(N);
	for(auto &a : A) cin >> a;
	ll l = A[0];
	for(int i=1;i<N;i++) {
		l = lcm(l, A[i]);
	}
	ll ans = 0;
	for(int i=0;i<N;i++) {
		ans += l/A[i];
		ans %= MOD;
	}
	cout << ans << endl;
}

File: abc152/e.py
MOD = int(1e9+7)

def gcd(a, b):
    if b == 0:
        return a
    return gcd(b, a%b)

def lcm(a, b):
    return a // gcd(a,b) * b


N = int(input())
A = [int(i) for i in input().split()]

l = 1
for i in range(N):
    l = lcm(l, A[i])

ans = 0
for i in range(N):
    ans += l // A[i]

print(ans % MOD)
